core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -702,61 +705,61 @@ class KafkaApis(val requestChannel: RequestChannel,
   /*
    * Handle an offset fetch request
    */
-
   def handleOffsetFetchRequest(request: RequestChannel.Request) {
-    val offsetFetchRequest = request.requestObj.asInstanceOf[OffsetFetchRequest]
+    val header = request.header
+    val offsetFetchRequest = request.body.asInstanceOf[OffsetFetchRequest]
 
+    val responseHeader = new ResponseHeader(header.correlationId)
+    val offsetFetchResponse =
     // reject the request immediately if not authorized to the group
```

Comment:
Its still "immediately". We generate the appropriate OffsetFetchResponse, return, and the next call sends it.